server/src/test/java/org/elasticsearch/action/admin/indices/shrink/TargetNumberOfShardsCalculatorTests.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.shrink;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.index.shard.DocsStats;
+import org.elasticsearch.index.store.StoreStats;
+import org.elasticsearch.test.ESTestCase;
+
+import static org.elasticsearch.action.admin.indices.shrink.TargetNumberOfShardsCalculator.Shrink.calculateAcceptableNumberOfShards;
+
+public class TargetNumberOfShardsCalculatorTests extends ESTestCase {
+
+    public void testShrink() {
+        Settings settings = Settings.builder()
+            .put("index.number_of_shards", randomIntBetween(1, 5))
+            .put("index.number_of_replicas", randomIntBetween(0, 5))
+            .put("index.version.created", Version.CURRENT)
+            .build();
+        IndexMetadata indexMetadata = IndexMetadata.builder("index").settings(settings).build();
+
+        TargetNumberOfShardsCalculator.Shrink shrink = new TargetNumberOfShardsCalculator.Shrink(
+            new StoreStats(between(1, 100), between(0, 100), between(1, 100)),
+            (i) -> new DocsStats(between(1, 1000), between(1, 1000), between(0, 10000))
+        );
+        assertEquals(4, shrink.calculate(4, null, indexMetadata));
+        assertEquals(1, shrink.calculate(null, null, indexMetadata));
```

Comment:
should this be illegal? ie. an exception thrown or such? It's not clear to me why this would be `1`

Reply:
Confirmed, this path is triggered but this request: `POST /many-shards/_shrink/shrinked`

Reply:
Ah thanks for checking this. I think we should improve the shrink API docs (not in this PR) to state that we default to 1 when no target (number of shards, or max primary shard size) is specified